tests/dragonfly/replication_test.py

```diff
@@ -1836,3 +1836,64 @@ async def test_replicaof_reject_on_load(df_local_factory, df_seeder_factory):
     await c_replica.close()
     master.stop()
     replica.stop()
+
+
+# note: please be careful if you want to change any of the parameters used in this test.
+# changing parameters without extensive testing may easily lead to weak testing case assertion
+# which means eviction may not get triggered.
+@pytest.mark.asyncio
+@pytest.mark.slow
+async def test_eviction_propagation(df_local_factory, df_seeder_factory):
+    master = df_local_factory.create(
+        proactor_threads=1, cache_mode="true", maxmemory="256mb", enable_heartbeat_eviction="false"
+    )
+    replica = df_local_factory.create(proactor_threads=1)
+    df_local_factory.start_all([master, replica])
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica)
+
+    print("Filling the master to approach maxmemory...")
+    seeder = df_seeder_factory.create(
+        port=master.port, keys=100000, val_size=1000, stop_on_failure=False
+    )
+
+    print("Filling data to trigger policy based eviction")
+    fill_task = asyncio.create_task(seeder.run())
+    while True:
+        time.sleep(0.3)
+        info = await c_master.info("stats")
+        if info["evicted_keys"] > 0:
+            break
+    fill_task.cancel()
+    await asyncio.sleep(5.0)
+
+    # Check data after stable state stream
+    await check_all_replicas_finished([c_replica], c_master)
+    await check_data(seeder, [replica], [c_replica])
+
+    # now enable heart beat eviction
+    info = await c_master.info("stats")
+    evict_after_first_test = info["evicted_keys"]
+    print("Turning on heart beat eviction and test again...")
+    await c_master.execute_command("CONFIG SET enable_heartbeat_eviction true")
+
+    await seeder.run(target_deviation=0.05)
```

Comment:
do we need to start seeder again to get the heartbeat eviction starting evicting? I think we dont
If so you can check the number of evicted before config set, than do config set , than sleep 2 seconds than check the number of evicted again and assert it is greater than the evicted before
after this you dont need to wait for heartbeat to stop evicting instead you can stop it again with config set

Reply:
in this current setting, we do. Because previous seeder run only uses about 150MB memory, which is far away from the hb eviction watermark. (e.g. 90% of maxmemory)